### PR TITLE
Fh 3830 - Error to create project, service and application after changes made into the templates

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/FH-3830

**Motivation**: Fix the PR https://github.com/feedhenry/fh-fhc/pull/369.

**Steps to verify**

fhc templates apps <template-id>
fhc templates projects <template-id>
fhc templates services <template-id>

fhc projects create --title=<title>
fhc services create --title=<title>
fhc apps create <projectId> --title=<title>

**Following the local tests.** 

To create a app
```
dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js app create k2654bvzrmaebx7ocnwtzxfp  --title="test4" --type=cloud_nodejs
{
  "scmCacheKey": "7cb34dd6479be4263d1acaf3089d6372",
  "hierarchy": null,
  "description": "",
  "domain": "support",
  "type": "cloud_nodejs",
  "openshift": null,
  "openshift3": null,
  "scmCommit": null,
  "jsonTemplateId": "hello_world_mbaas_instance",
  "template": null,
  "email": "cmacedo@redhat.com",
  "sysOwner": "i4ze7ruomd27wnohth22to4v",
  "guid": "k2654btv5ipfwlf4apfc4c4x",
  "title": "test4",
  "sysModified": 1501788889359,
  "legacy": false,
  "gitRefs": {},
  "sysVersion": 1,
  "sourcePath": "",
  "embed": false,
  "gitApp": true,
  "deployedGitRefs": {},
  "sysCreated": 1501788888058,
  "scmUrl": "git@git.us.feedhenry.com:support/01887836POC-test4-1.git",
  "scmKey": null,
  "scmBranch": "master",
  "internallyHostedRepo": true,
  "client": false,
  "newApp": true,
  "targetable": true,
  "apiKey": "12020279f8719af2fc083900169e98a906b6edb1",
  "windowsPhonePushInformation": null,
  "androidPushInformation": null,
  "iOSPushInformation": null,
  "internallyHostedRepoUrl": "git@git.us.feedhenry.com:support/01887836POC-test4-1.git",
  "buildJobName": "01887836POC-test4-1_k2654btv5ipfwlf4apfc4c4x",
  "sysGroupFlags": 65567,
  "sysGroupList": "",
  "businessObject": "cluster/reseller/customer/domain/project/cloud-apps",
  "deployable": true,
  "envsToDeployOnCreate": [],
  "envsToDeployOnUpdate": [],
  "initialisedFromTemplate": "true",
  "migrated": false,
  "internallyHostedRepoName": "01887836POC-test4-1",
  "buildCredentialName": "01887836POC-test4-1_k2654btv5ipfwlf4apfc4c4x-gitAccessKey",
  "pushInfo": {},
  "hasOwnDb": {
    "openshift": false,
    "dev": false,
    "sp1-openshift": false,
    "live": false
  },
  "wrapperModule": {
    "openshift": "fh-nodeapp",
    "dev": "fh-nodeapp",
    "sp1-openshift": "fh-nodeapp",
    "live": "fh-nodeapp"
  },
  "sourcePathConfigurable": false,
  "previewable": false,
  "buildable": false,
  "runtime": {
    "openshift": null,
    "dev": null,
    "sp1-openshift": null,
    "live": null
  }
}
dhcp-17-108:fh-fhc cmacedo$ 
```

To create a project 

```
dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js projects create --title="test4" 
Project 'test4' create successfully
```

```
dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js projects create --title="test44" --json 
{
  "description": "",
  "type": "PROJECT",
  "templateId": "default",
  "jsonTemplateId": "bare_project",
  "template": null,
  "sysOwner": "i4ze7ruomd27wnohth22to4v",
  "guid": "ypqw2fczuh44uk3r5vx6o4ht",
  "title": "test44",
  "sysModified": 1501788967766,
  "sysVersion": 0,
  "featured": false,
  "sysCreated": 1501788967765,
  "apps": [],
  "authorEmail": "cmacedo@redhat.com",
  "sysGroupFlags": 65567,
  "sysGroupList": "",
  "businessObject": "cluster/reseller/customer/domain/project",
  "migrated": false,
  "service": false
}
```

To create a service

```
dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js services create --title="test4" 
Service 'test4' create successfully
```

```
dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js services create --title="test44" --json 
{
  "description": "",
  "type": "CONNECTOR",
  "templateId": "other",
  "jsonTemplateId": "new-service",
  "template": null,
  "sysOwner": "i4ze7ruomd27wnohth22to4v",
  "guid": "rrez7bgfe7h4w2g4vhuno7e4",
  "title": "test44",
  "sysModified": 1501788975148,
  "sysVersion": 0,
  "featured": false,
  "sysCreated": 1501788975147,
  "apps": [
    {
      "scmCacheKey": "b3c8c6b90be2ce714c03de6668bfccb9",
      "hierarchy": null,
      "description": "",
      "domain": "support",
      "type": "cloud_nodejs",
      "openshift": null,
      "openshift3": null,
      "scmCommit": null,
      "jsonTemplateId": "hello_world_mbaas_instance",
      "template": null,
      "email": "cmacedo@redhat.com",
      "sysOwner": "i4ze7ruomd27wnohth22to4v",
      "guid": "rrez7bh4zefewlaq5yvb7npi",
      "title": "test44",
      "sysModified": 1501788977355,
      "legacy": false,
      "gitRefs": {},
      "sysVersion": 1,
      "sourcePath": "",
      "embed": false,
      "gitApp": true,
      "deployedGitRefs": {},
      "sysCreated": 1501788975178,
      "scmUrl": "git@git.us.feedhenry.com:support/test44-test44.git",
      "scmKey": null,
      "scmBranch": "master",
      "internallyHostedRepo": true,
      "client": false,
      "newApp": true,
      "targetable": true,
      "apiKey": "24ff144763ec9703e43a8f3d05c4c892e1637214",
      "windowsPhonePushInformation": null,
      "androidPushInformation": null,
      "iOSPushInformation": null,
      "internallyHostedRepoUrl": "git@git.us.feedhenry.com:support/test44-test44.git",
      "buildJobName": "test44-test44_rrez7bh4zefewlaq5yvb7npi",
      "sysGroupFlags": 65567,
      "sysGroupList": "",
      "businessObject": "cluster/reseller/customer/domain/service/cloud-apps",
      "deployable": true,
      "envsToDeployOnCreate": [],
      "envsToDeployOnUpdate": [],
      "initialisedFromTemplate": "true",
      "migrated": false,
      "internallyHostedRepoName": "test44-test44",
      "buildCredentialName": "test44-test44_rrez7bh4zefewlaq5yvb7npi-gitAccessKey",
      "pushInfo": {},
      "hasOwnDb": {
        "openshift": false,
        "dev": false,
        "sp1-openshift": false,
        "live": false
      },
      "wrapperModule": {
        "openshift": "fh-nodeapp",
        "dev": "fh-nodeapp",
        "sp1-openshift": "fh-nodeapp",
        "live": "fh-nodeapp"
      },
      "sourcePathConfigurable": false,
      "previewable": false,
      "buildable": false,
      "runtime": {
        "openshift": null,
        "dev": null,
        "sp1-openshift": null,
        "live": null
      }
    }
  ],
  "authorEmail": "cmacedo@redhat.com",
  "sysGroupFlags": 65567,
  "sysGroupList": "",
  "businessObject": "cluster/reseller/customer/domain/service",
  "migrated": false,
  "service": true
}
dhcp-17-108:fh-fhc cmacedo$ 
```

### CODE REVIEW

**Comment by camilamacedo86** [comment]
@mmusil could you test it for us and approve the PR?

File: lib/cmd/fh2/app/create.js [hunk 1]
```diff
@@ -106,18 +106,14 @@ function doCreateNguiFromTemplate(projectId, title, templateId, deployEnv, cb) {
...
-  fhc.templates.apps({id:templateId}, function(err, template) {
+  fhc.templates.apps({id:templateId, json:true}, function(err, template) {
     if (err) {
       return cb(err);
     }
-    if (!template) {
-      return cb(i18n._('Template not found: ') + templateId);
-    }
-    payload.template = template;
+    payload.template = template[0];
```
Comment (wei-lee) [inline lib/cmd/fh2/app/create.js]: better check `template` is an array. Also might be better rename it to `templates` instead

**Comment by camilamacedo86** [comment]
@wei-lee :-) Tks

File: lib/cmd/fh2/app/create.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/cmd/fh2/app/create.js]: It will be removed :-) into the https://github.com/feedhenry/fh-fhc/pull/342

**Pull request merged**